02_commands.html

```diff
@@ -1,206 +1,524 @@
-          object.
-          <ol>
-            <li>If <a href="#newsession">newSession()</a> is called then the response payload body would look like the following
-              <pre>
-                {
-                  "sessionId" : "string value",
-                  "value" : capabilities-object,
-                }.
-              </pre>
-            </li>
-            <li>
-              If the WebDriver Command has to return data it should be the whole of the response payload body and look like the following
-              <pre>
-                {
-                  "value": "data from WebDriver Commands"
-                }
-              </pre>
-            </li>
-            <li>
-              If the WebDriver Command results in an error then the response payload body would look like the following
-              <pre>
-                {
-                  "status": "string value",
-                  "value" : "string message"
-                }
-              </pre>
-            </li>
-          </ol>
-        </li>
-        <li>Set HTTP headers for the follow keys and their values.
-          <ul>
-            <li><code>cache-control: no-cache</code></li>
-            <li><code>content-type: application/json</code></li>
-          </ul>
-        </li>
-        <li>Return HTTP response to the Local End</li>
-      </ol>
-    </section>
+    <h3>Introduction</h3>
+    <p>TODO</p>
   </section>
   <section>
-    <h3>Status Codes</h3>
-    <p>The WebDriver API indicates the success or failure of a command invocation via <var>status</var>. The following values are used and have the following meanings.</p>
+    <h3>Common Infrastructure</h3>
+    <h4>Algorithms</h4>
+    <p>Various parts of this specification are written in terms of step-by-step algorithms. The details of these algorithms do not have any normative significance; implementations are free to adopt any implementation strategy that produces equivalent output to the specification. In particular algorithms in this document are optimised for readability rather than performance.</p>
+    <p>Where algorithms that return values are fallible, they are written in terms of returning either <dfn>success</dfn> or <dfn>error</dfn>. A <a>success</a> value has an associated <var>data</var> field which encapsulates the value returned, whereas an <a>error</a> response has an associated <a>error code</a>.</p>
+  </section>
+
+  <section>
+    <h3>Processing Model</h3>
+    <p>The <a>remote end</a> acts as a HTTP server reading requests from the client and writing responses, typically over TCP socket. For the purposes of this specification we model the data transmission between a particular <a>local end</a> and <a>remote end</a> with a <dfn>connection</dfn> object to which the <a>remote end</a> may <dfn>write bytes</dfn> and <dfn>read bytes</dfn>. However the exact details of how this connection works and how it is established are out of scope.</p>
```

Comment:
Drop “object” from “connection object”; I think we can just talk about this thing that is a read-writer stream.